plugins/usd/module/vtkF3DUSDImporter.cxx

```diff
@@ -290,6 +409,15 @@ class vtkF3DUSDImporter::vtkInternals
                   // interpolation is face-varying.
                   // It will be normalized by the vtkF3DFaceVaryingPointDispatcher later
                   newPolyData->GetPointData()->AddArray(texCoords);
+
+                  if (firstArray)
+                  {
+                    // sometimes we are enable to fetch the array name to use for texture mapping
+                    // so we fallback to the first UV set added
+                    // see https://github.com/f3d-app/f3d/issues/1184
+                    firstArray = false;
+                    newPolyData->GetPointData()->SetTCoords(texCoords);
```

Comment:
Ok I get it now. You set it here in case its not set later ?

Reply:
Yes it can be empty in case we have a texture transform layer, but we do not support it yet.

Reply:
Yes it can be empty in case we have a texture transform layer, but we do not support it yet.